pymongo/database.py

```diff
@@ -783,7 +784,7 @@ def list_collections(
         filter: Optional[Mapping[str, Any]] = None,
         comment: Optional[Any] = None,
         **kwargs: Any,
-    ) -> CommandCursor[Dict[str, Any]]:
+    ) -> CommandCursor[MutableMapping[str, Any]]:
```

Comment:
Why change from Dict to MutableMapping?

Reply:
I removed `Dict` from `_DocumentType` to align with `_DocumentOut`, which meant this method and `list_databases` had to be changed as well.

Reply:
But dict is a MutableMapping so why can't we return it? Is there a mypy error preventing it?

Reply:
`Dict` is more specific than `MutableMapping`.  We can either put `Dict` back in `_DocumentType` or use `MutableMapping` here.

Reply:
```
pymongo/database.py: note: In member "list_collections" of class "Database":
pymongo/database.py:787: error: Value of type variable "_DocumentType" of "CommandCursor" cannot be "dict"  [type-var]
        ) -> CommandCursor[Dict[str, Any]]:
             ^
Found 1 error in 1 file (checked 55 source files)
```

I think this is a bug. It has to be possible to use Dict/dict as a generic for cursor/collection/db/client otherwise the generic feature is not useful:
```
$ cat generic.py 
from typing import Dict, Any
from pymongo import MongoClient

client: MongoClient[Dict[str, Any]] = MongoClient()
doc = client.test.test.find_one()
if doc is not None:
    doc['a'] = 1
$ python generic.py 
$ mypy generic.py 
generic.py:4: error: Value of type variable "_DocumentType" of "MongoClient" cannot be "dict"  [type-var]
    client: MongoClient[Dict[str, Any]] = MongoClient()
            ^
generic.py:4: error: Incompatible types in assignment (expression has type "MongoClient[MutableMapping[str, Any]]", variable has type
"MongoClient[Dict[str, Any]]")  [assignment]
    client: MongoClient[Dict[str, Any]] = MongoClient()
                                          ^
Found 2 errors in 1 file (checked 1 source file)
```

Reply:
Yeah, I think this is related to https://github.com/python/mypy/issues/3737.  I say we do not try and make it generic on the input value